### ISSUE TITLE
POST ... 400 Bad Request

### ISSUE THREAD

**Issue by ItzArty** [socketio/socket.io-client#1422]
Hello, I was recently facing a problem, where my server crashed after disconnecting, I was running version 1.0.x and the rest worked pretty much fine. After upgrading my socket.io module, there were problems, problems and problems. Firstly I hit the CORS, I partitialy solved that, adding the credential thing on both client and server, but now I receive the 400 Bad Request error. Dude, f**k. Now, I tried even adding custom headers and things but noooooothing works. Im new to the socket.io and literally, this drives me nuts. I want to further develop my project but no, fk me. So, please. What am I just supposed to? I did open the url in browser, that was 400ed and it had a normal return. Why the hell then my website cannot access it. This happens even on localhost, however, yes, if youre asking, I have my website under cloudflare, but I disabled HTTPS so that should not be the case. So, PLEASE, help me.
Here is some of my code:

server.js:
```
var io = require('socket.io')(3090, {
	cors: {
		origin: "http://www.my.domain",
		methods: ["GET", "POST"],
		allowedHeaders: ["my-custom-header"],
		credentials: true
	}
});
```

client.js:
`	socket = io.connect('http://servers.numeric.ip:' + port, { withCredentials: true, extraHeaders: { "my-custom-header": "abcd" } });
`

**Comment by loczek2001** [comment]
The client has the 2.0 version of socket io and it doesn't work with the server 3.0 version so make sure to upgrade the client version.

**Comment by matiaslopezd** [comment]
@ItzArty sorry but the explanation does not have sense around disconnect socket event, that crash your app. How @loczek2001 tells you, to check your versions and ensure that the major version is the same on the server and client side. Also, check that the function argument you pass to the disconnect event is working, add try/catch to check what happens.

About Cloudflare, only need to check that the option of allow the websockets connections is activated. SSL issues, browsers return other code, not 400.

**Comment by ItzArty** [comment]
I'm sorry for not closing the thread!
I have discovered that for some reason, when I used NPM to update socket io it seemed to update alrighty, but, then I could do this infinitely, update and update and update... Then I discovered that NPM downloaded the update, but for some reason didn't replace files of the old version, therefore never updated. Now that I copied the temp files and pasted them, everything works properly.
